### PR TITLE
Update style spec

### PR DESCRIPTION
**Pull request by orangemug**
I'm a little unsure this is a 100% working PR, mainly because we're missing good test coverage, which will probably be my next task. When reviewing this PR can someone checkout the branch and give it a test locally.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -30,9 +30,8 @@
     "lodash.clonedeep": "^4.5.0",
     "lodash.isequal": "^4.4.0",
     "lodash.throttle": "^4.1.1",
-    "mapbox-gl": "^0.33.0",
-    "mapbox-gl-inspect": "^1.2.2",
-    "mapbox-gl-style-spec": "^8.11.0",
+    "mapbox-gl": "^0.34.0",
+    "mapbox-gl-inspect": "lukasmartinelli/mapbox-gl-inspect#7108dab80f",
```
Comment (orangemug) [inline package.json]: This points at the git commit because the latest version of mapbox-gl-inspect hasn't been released to npm yet.

Comment (lukasmartinelli) [inline package.json]: lemme release that.

Comment (lukasmartinelli) [inline package.json]: `mapbox-gl-inspect@1.2.3` published

**Comment by lukasmartinelli** [comment]
Tried it out!! Works well. Merge master and then merge the PR!

giphy 19

**Comment by orangemug** [comment]
Ah I see we should be using https://www.npmjs.com/package/@mapbox/mapbox-gl-style-spec however that is in ES6 so we need to precompile

File: config/webpack.loaders.js [hunk 2]
```diff
@@ -6,7 +6,7 @@ module.exports = [
   },
   {
     test: /\.jsx?$/,
-    exclude: /(node_modules|bower_components|public)/,
+    exclude: /(.*node_modules(?!\/@mapbox\/mapbox-gl-style-spec)|bower_components|public)/,
```
Comment (orangemug) [inline config/webpack.loaders.js]: Now compiling `@mapbox/mapbox-gl-style-spec`

**Comment by lukasmartinelli** [comment]
Uggh sorry it is so much trouble. I am not the biggest fan of the monorepo of GLJS :cry:

**Comment by orangemug** [comment]
That moment where you forget you modified [@mapbox/mapbox-gl-style-spec/migrate/v8.js](https://github.com/mapbox/mapbox-gl-js/blob/master/src/style-spec/migrate/v8.js#L3) to reference `.json` to fix the tests...

**Comment by orangemug** [comment]
On hold waiting for PR to be merged and npm module released <https://github.com/mapbox/mapbox-gl-js/pull/4563>

**Comment by orangemug** [comment]
This PR should be passing the tests now, but need further consideration as `@mapbox/mapbox-gl-style-spec` has been bumped to v9, so it needs a good manual test run. I'm also waiting on a response to https://github.com/mapbox/mapbox-gl-js/pull/4563#issuecomment-293815581 so please don't merge this PR until the above are complete.

**Comment by orangemug** [comment]
Well this is a fun PR... My best guess for the failing tests on windows is that this regexp isn't matching therefore we aren't building the @mapbox/mapbox-gl-style-spec (see https://github.com/orangemug/editor/blob/c552838fddb41ad3ba2d8cf4187a860865933a98/config/webpack.loaders.js#L9). That is kinda of a hack anyway because babel shouldn't need to touch that module, however the built module isn't ES5... I'll find a windows box later and try to diagnose, but it might be better to raise a few issues on @mapbox/mapbox-gl-style-spec repo and wait for the fixes.

**Comment by lukasmartinelli** [comment]
Yeah one would think JavaScript is cross platform.. but I ran into many troubles with Linux/OSX/Windows

**Comment by orangemug** [comment]
I currently don't have access to a windows machine. If anyone would like to take a look into the windows issue please feel free, see <https://github.com/maputnik/editor/pull/124#issuecomment-293885705> for details

**Comment by orangemug** [comment]
I've found the issue, see <https://github.com/webpack/webpack/issues/2073>

**Pull request merged**